packages/assets-controllers/src/TokenRatesController.ts

```diff
@@ -203,7 +214,9 @@ export class TokenRatesController extends BaseController<
       interval: 3 * 60 * 1000,
       nativeCurrency: initialTicker,
       chainId: initialChainId,
-      tokens: [],
+      selectedAddress: initialSelectedAddress,
+      allTokens: {}, // TODO: initialize these correctly, maybe as part of BaseControllerV2 migration
```

Comment:
I'm just now noticing that the only thing that TokenRatesController has in state is `contractExchangeRates`. Is `config` persisted to state and if not, shouldn't it at least be read-only? i.e. should `allTokens` and `allDetectedTokens` be a part of state too?

Reply:
Config and state are separate; config is not persisted to-disk.

I'm not sure I follow the question about whether it should be read-only; are you talking about config or state there? What exactly should we consider making read-only and why?

I don't understand why we'd move `allTokens` and `allDetectedTokens` into state; they're already in the state of the tokens controller. That would cause an overlap.

Reply:
By read-only, I mean that this property is only used to set the initial private state of the controller and is not changed. It seems that we are changing `allTokens` and `allDetectedTokens` throughout the lifetime of the controller, however, and that we'd want to instead use the values that come through in `config` to set private properties that then get changed.

Perhaps this pattern is not new, though, and `config` has been used as form of private state all along. I suppose it's not a big deal — this will go away when the controller is converted to BaseControllerV2 — I just didn't know if it was intentional or not.

Reply:
Indeed, this is how config has always worked unfortunately. Agreed that it's not good. That will be fixed with the BaseControllerV2 migration